src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -104,13 +105,115 @@ String getLogStreamName(String thingName) {
         return logStreamName.replace(":", "+");
     }
 
+    /**
+     * Returns the name of the CloudWatch log group.
+     *
+     * @param componentType ComponentType
+     * @param componentName Name of the component
+     */
+    public String getLogGroupName(ComponentType componentType, String componentName) {
+        String awsRegion = Coerce.toString(deviceConfiguration.getAWSRegion());
+
+        return DEFAULT_LOG_GROUP_NAME
+                .replace("{componentType}", componentType.name())
+                .replace("{region}", awsRegion)
+                .replace("{componentName}", componentName);
+    }
+
+    /**
+     * Gets CW input log events from the component which processLogFiles need to be uploaded to CloudWatch.
+     *
+     * @param  attempt a cloudwatch attempt
+     * @param cpInfo log files information for a component to read logs from.
+     * @param fileInfo metadata of the file
+     * @param chan a byte channel that allows to read a file from the specified position
+     *
+     * @throws IOException if it fails to read the file or the file has been deleted
+     */
+    public CloudWatchAttempt processLogFiles(
+            CloudWatchAttempt attempt, ComponentLogFileInformation cpInfo, LogFileInformation fileInfo,
+            SeekableByteChannel chan) throws IOException {
+        long startPosition = fileInfo.getStartPosition();
+        boolean reachedMaxSize = false;
+        AtomicInteger totalBytesRead = new AtomicInteger();
```

Comment:
do you know why this is an atomic integer? was the previous code being called in parallel?

Reply:
It was an atomic integer before because it was being called inside a lambda and given it is being passed on `processLogLine` and then to other methods I didn't want to modify those methods yet and bloat this even more.

Reply:
Not sure. Theoratically, this is only one thread.

Reply:
As I replied. The only reason for this to be an atomic integer was before before this was inside of lambda and all the methods that where called inside of a lambda expect an atomic integer on their signature. We can change it but I did not in the spirit of not adding more changes. Let's table changing this for the next PR unless you folks think otherwise